sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -247,6 +247,22 @@ object functions {
   def last(columnName: String): Column = last(Column(columnName))
 
   /**
+   * Aggregate function: returns the average of the values in a group.
+   * Alias for avg.
+   *
+   * @group agg_funcs
+   */
+  def mean(e: Column): Column = avg(e)
+
+  /**
+   * Aggregate function: returns the average of the values in a group.
+   * Alias for avg.
+   *
+   * @group agg_funcs
+   */
+  def mean(columnName: String): Column = avg(columnName)
```

Comment:
So this is weird. It already seems to exist in Python https://github.com/apache/spark/blob/35d6a99cbe3f67da5d56888e63baf9bc69f3de91/python/pyspark/sql/functions.py#L117 -- Not sure if it was just broken before